### PR TITLE
Alt-click on disposal bins to manually eject their contents

### PR DESCRIPTION
**Pull request by Adrer**
## What Does This PR Do
Allows humanoids (Borgs are unable to) to Alt-click on disposal bins to manually eject their contents after 5 seconds, even when depowered.

## Why It's Good For The Game
Currently when a disposals bin is depowered, the only way to get the contents out is by destroying the bin or powering it. This PR should offer a viable alternative to this.

## Testing
Put something in a disposals bin, alt-click it, 5 seconds later the contents are ejected.

## Changelog
:cl:
tweak: Alt-clicking on a disposal bin as a human now manually ejects its contents.
/:cl:

### CODE REVIEW

File: code/modules/recycling/disposal.dm [hunk 1]
```diff
@@ -317,6 +317,18 @@
...
 
+/obj/machinery/disposal/AltClick(mob/user)
+	if(!Adjacent(user) || !ishuman(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
+		return
+	user.visible_message(
+	"<span class='notice'>[user] tries to eject the contents of [src] manually.</span>",
+	"<span class='notice'>You operate the manual ejection lever on [src].</span>")
+	if(do_after(user, 5 SECONDS, target = src))
+		user.visible_message(
+		"<span class='notice'>[user] ejects the contents of [src].'.</span>",
```
Comment (lewcc) [inline code/modules/recycling/disposal.dm:328]:
```suggestion
		"<span class='notice'>[user] ejects the contents of [src].</span>",
```

**Comment by matttheficus** [comment]
Im assuming the cooldown is to prevent Officers/Paramedics/etc from one-click checking for bin'd bodies?

**Comment by Adrer** [comment]
> Im assuming the cooldown is to prevent Officers/Paramedics/etc from one-click checking for bin'd bodies?

Yeah, it's not supposed to be a replacement to actually ejecting bins via the interface, it's mostly intended to allow you to eject the contents of a depowered bin.

**Comment by zzzmike** [comment]
Very good idea. If this could be added to other appropriate things like body scanners in medical (I ran into a depowered body scanner thing recently) that would be cool too.

**Comment by Adrer** [comment]
> Very good idea. If this could be added to other appropriate things like body scanners in medical (I ran into a depowered body scanner thing recently) that would be cool too.

Body scanners/sleepers already have this functionality, actually.

File: code/modules/recycling/disposal.dm [hunk 2]
```diff
@@ -317,6 +317,18 @@
...
+	if(!Adjacent(user) || !ishuman(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
+		return
+	user.visible_message(
+	"<span class='notice'>[user] tries to eject the contents of [src] manually.</span>",
+	"<span class='notice'>You operate the manual ejection lever on [src].</span>")
+	if(do_after(user, 5 SECONDS, target = src))
+		user.visible_message(
+		"<span class='notice'>[user] ejects the contents of [src].</span>",
+		"<span class='notice'>You eject the contents of [src].</span>")
+		eject()
```
Comment (lewcc) [inline code/modules/recycling/disposal.dm:330]:
```suggestion
	user.visible_message(
		"<span class='notice'>[user] tries to eject the contents of [src] manually.</span>",
		"<span class='notice'>You operate the manual ejection lever on [src].</span>"
	)
	if(do_after(user, 5 SECONDS, target = src))
		user.visible_message(
			"<span class='notice'>[user] ejects the contents of [src].</span>",
			"<span class='notice'>You eject the contents of [src].</span>"
		)
		eject()
```

**Review by lewcc — COMMENTED** [review]
Code looks good, just one nitpick with indent style.

**Review by farie82 — APPROVED** [review]

**Pull request merged**